src/main.c

```diff
@@ -150,13 +151,39 @@ static void create_calendars(FocalApp* fm)
 	}
 }
 
+static void update_header_subtitle(FocalApp* fm)
+{
+	WeekView* wv = FOCAL_WEEK_VIEW(fm->weekView);
+
+	struct tm day_start = week_view_get_start(wv);
+	char start[24];
+	strftime(start, sizeof(start), "%e. %B %G", &day_start);
+
+	struct tm day_end = week_view_get_end(wv);
+	char end[24];
+	day_end.tm_hour -= 2;
+	mktime(&day_end);
+
+	strftime(end, sizeof(end), "%e. %B %G", &day_end);
+
+	char title[52];
+	snprintf(title, 96, "%s – %s", start, end);
+
+	gtk_header_bar_set_subtitle(GTK_HEADER_BAR(fm->header), title);
+}
+
 static void update_window_title(FocalApp* fm)
 {
-	int week_num = week_view_get_current_week(FOCAL_WEEK_VIEW(fm->weekView));
-	char week_title[8];
-	snprintf(week_title, 8, "Week %d", week_num);
+	WeekView* wv = FOCAL_WEEK_VIEW(fm->weekView);
+
+	int week_num = week_view_get_week(wv);
 
-	gtk_window_set_title(GTK_WINDOW(fm->mainWindow), week_title);
+	char title[8];
+	snprintf(title, 8, "Week %d", week_num);
+
+	gtk_window_set_title(GTK_WINDOW(fm->mainWindow), title);
+
+	update_header_subtitle(fm);
```

Comment:
If there is never any situation where `update_header_subtitle` can be called independently of `update_window_title` then I prefer to use a single function (with comments if necessary marking the title and subtitle parts)